localstack-core/localstack/services/lambda_/provider.py

```diff
@@ -1866,6 +1869,39 @@ def update_alias(
     # =======================================
     # ======= EVENT SOURCE MAPPINGS =========
     # =======================================
+    def check_service_resource_exists(self, service, resource_arn):
+        arn = parse_arn(resource_arn)
+        source_client = get_internal_client(resource_arn)
```

Comment:
We would need to create the internal client with the same kind of parameters that we create it in the actual event source mapping:
```python
source_client = get_internal_client(
    arn=resource_arn,
    role_arn=<configured-lambda-role-arn>,
    service_principal=ServicePrincipal.lambda_,
    source_arn=<lambda-function-arn>,
)
```

This way, we properly pass required metadata about the origin and the original caller of the inter-service request. You will need the lambda function version role, which is fetched a bit later after validation, so you might need to do a bit of refactoring, and the lambda function ARN.